sqlmesh/core/engine_adapter/redshift.py

```diff
@@ -54,6 +54,18 @@ def create_view(
             view_name, query_or_df, columns_to_types, replace, no_schema_binding=True
         )
 
+    def columns(self, table_name: TableName) -> t.Dict[str, exp.DataType]:
+        """Fetches column names and types for the target table."""
+        table = exp.to_table(table_name)
+        sql = (
+            exp.select("column_name", "data_type")
+            .from_("SVV_COLUMNS")
+            .where(f"table_name = '{table.alias_or_name}' AND table_schema = '{table.args['db']}'")
```

Comment:
```suggestion
            .where(f"table_name = '{table.alias_or_name}' AND table_schema = '{table.db}'")
```

Reply:
Actually, there's a difference: what you currently have will implicitly invoke `table.args['db'].sql()` without a dialect because it's interpolated in the string, whereas in my suggestion you just get the raw text of the `db` field. These two are essentially different when `db` is quoted -- not sure if this can happen though.

Reply:
Yes, it can be quoted. Which is the correct syntax to maintain quoting?

Reply:
I guess I don't want to maintain quoting because in the where clause the quotes are added. So raw string is better?

Reply:
Sounds reasonable

Reply:
(FWIW if you wanted to preserve the quotes you could just do `table.args['db'].sql(dialect=self.dialect)`)